source/features/hide-inactive-deployments.js

```diff
@@ -0,0 +1,22 @@
+import select from 'select-dom';
+import observeEl from '../libs/simplified-element-observer';
+
+export default function () {
+	observeEl('.js-discussion', () => {
+		const deployments = select.all('.discussion-item .deployment-meta');
+		deployments.pop(); // Don't hide the last deployment, even if it is inactive
+
+		for (const deployment of deployments) {
+			const isInactiveDeployment = select.exists('.is-inactive', deployment);
+			if (isInactiveDeployment) {
+				const discussionItem = deployment.closest('.discussion-item');
+				discussionItem.dataset.rghInactiveDeployment = 'true';
+			}
+		}
+	}, {
+		childList: true,
+		attributes: true,
+		subtree: true,
+		attributeFilter: ['class']
```

Comment:
Is this all necessary? I think others observers on the discussion only listen to childList. Subtree is heavy

Reply:
I'm not sure if the node list is updated when a deployment changes from active to inactive. That's why I'm watching class names.

Reply:
I think you're still selecting them all, so whether the class is there or not will be found by `is-inactive`, right?

Worst case scenario there's an old inactive deployment on the current page... because the user never left the page. I'll take that over watching all nodes on PR pages.

Reply:
Just tested it again at FloEdelmann/open-fixture-library#449 with only `childList` set to true:

1. Initial timeline: (only commits and deployments)
   - 2 commits
   - Active deployment (with `View deployment` button)
2. Directly after pushed commit: 
   - 2 commits
   - Active deployment (with `View deployment` button)
   - 1 commit
3. Some seconds later: 
   - 2 commits
   - Active deployment (with `View deployment` button)
   - 1 commit
   - Pending deployment (with `Pending` label)
4. About a minute later: 
   - 2 commits
   - 1 commit
   - Active deployment (with `View deployment` button)

I thought that at least `subtree` had to be turned on, as from 3. to 4., no element is (visually) added or removed from the timeline. However, it seems like that not the last deployment's contents are changed from pending to active, but the complete timeline item is replaced with a new node.

So you're right, `childList` is sufficient (as it's the default, I'll remove the whole options object).